### PR TITLE
Fix data type exceptions (#4)

### PR DESCRIPTION
**Pull request by AdwaithaV**
#UNITARY HACK -Fix data type exceptions (#4)

This PR fixes the issue where integer-valued parameters caused JAX gradient errors during optimization by:

- Automatically converting integer inputs to float32
- Adding warning messages when conversion occurs
- Ensuring all numeric parameters are JAX-compatible

This reopens the Unitary Hack - Fix data type exceptions #4 (#7)

### CODE REVIEW

File: src/qurveros/optspacecurve.py [hunk 1]
```diff
@@ -275,56 +275,52 @@ def barq_curve(x, params):
...
+            init_free_points = init_free_points.astype(jnp.float32)
+        elif not jnp.issubdtype(init_free_points.dtype, jnp.floating):
+            init_free_points = init_free_points.astype(jnp.float32)  # Fallback for other types
+
+    # Ensure pgf_params and prs_params use floating-point if they contain numeric values
+    if init_pgf_params is None:
+        init_pgf_params = barqtools.get_default_pgf_params_dict()
+    else:
+        init_pgf_params = {
+            k: v.astype(jnp.float32) if isinstance(v, jnp.ndarray) else v
```
Comment (gderossi) [inline src/qurveros/optspacecurve.py:307]: This covers the case where the parameter is a JAX array, but not other cases. Parameters can be scalars or arrays (which could take the form of Python lists or JAX arrays), and so any of those cases should be handled appropriately. Don't worry about any more complex data structures (nested dictionaries, custom classes, etc.).

File: src/qurveros/optspacecurve.py [hunk 2]
```diff
@@ -365,4 +361,4 @@ def save_bezier_control_points(self, filename):
         ])
 
         df = pd.DataFrame(points)
-        df.to_csv(filename, index=False, float_format='%.12f')
+        df.to_csv(filename, index=False, float_format='%.12f')
```
Comment (gderossi) [inline src/qurveros/optspacecurve.py:364]: Using a newline at the end of a file is good Python style! Please leave it in :)

**Comment by AdwaithaV** [comment]
@gderossi  Thank you for the suggestions i will create a new pr with necessary changes

**Comment by AdwaithaV** [comment]
Hi @gderossi,
I've created a new pull request incorporating the feedback you provided. Whenever you have time, I’d really appreciate it if you could take a look and let me know if any further changes are needed.
Thank you!
